File: frontend/src/api/adminDashboard.ts
import { apiClient } from "./client";

export interface AdminDashboardSummary {
  total_users: number;
  users_by_role: Record<string, number>;
  risk_distribution: Record<string, number>;
  prescription_status: Record<string, number>;
  review_stats: Record<string, number>;
  r2_review_rate: number;
  r3_referral_count: number;
  feedback_stats: {
    total: number;
    average_completion_rate: number;
  };
  template_usage: {
    total_templates: number;
    approved_templates: number;
  };
  cluster_distribution: Record<string, number>;
}

export function getAdminDashboardSummary() {
  return apiClient.get<AdminDashboardSummary>("/admin/dashboard/summary").then((response) => response.data);
}

File: backend/app/schemas/admin_dashboard.py
from pydantic import BaseModel


class FeedbackStats(BaseModel):
    total: int
    average_completion_rate: float


class TemplateUsageStats(BaseModel):
    total_templates: int
    approved_templates: int


class AdminDashboardSummary(BaseModel):
    total_users: int
    users_by_role: dict[str, int]
    risk_distribution: dict[str, int]
    prescription_status: dict[str, int]
    review_stats: dict[str, int]
    r2_review_rate: float
    r3_referral_count: int
    feedback_stats: FeedbackStats
    template_usage: TemplateUsageStats
    cluster_distribution: dict[str, int]
